### PR TITLE
Add a feature to view comments

### PR DESCRIPTION
**Pull request by doko-desuka**
This adds menu "Comments" to the "More..." dialog from the context menu. It takes you to a directory showing the "top level" comments made on the video, with the comment text as the plot in each item.  
The items are folders, entering a top-level comment will show the replies inside.

ytComments0

ytComments

### CODE REVIEW

**Comment by doko-desuka** [comment]
One thing I noticed is that the automatic pagination from the add-on is not kicking in. If I enter a comment with 457 replies it only shows 50, with no "next page" item.

**Comment by mrmvergeer** [comment]
I think that's an API restriction:  only the last 50 comments

File: resources/lib/youtube_plugin/youtube/helper/utils.py [hunk 1]
```diff
@@ -43,6 +43,27 @@ def extract_urls(text):
 
 def get_thumb_timestamp(minutes=15):
     return str(time.mktime(time.gmtime(minutes * 60 * (round(time.time() / (minutes * 60))))))
+    
+    
+def make_comment_item(snippet, uri, total_replies=0):
+    label = '[B]{}[/B]'.format(kodion.utils.to_utf8(snippet['authorDisplayName']))
+    if (snippet['publishedAt'] != snippet['updatedAt']):
+        label += ' (edited)'
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/youtube/helper/utils.py:51]: `edited` needs to be localized

File: resources/lib/youtube_plugin/youtube/helper/utils.py [hunk 2]
```diff
@@ -43,6 +43,27 @@ def extract_urls(text):
...
+    
+    extra_info = [ ]
+    if snippet['likeCount']:
+        extra_info.append('Likes: {likes:d}'.format(likes=snippet['likeCount']))
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/youtube/helper/utils.py:55]: `Likes` needs to be localized

File: resources/lib/youtube_plugin/youtube/helper/utils.py [hunk 3]
```diff
@@ -43,6 +43,27 @@ def extract_urls(text):
...
+    if total_replies:
+        extra_info.append('Replies: {total_replies:d}'.format(total_replies=total_replies))
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/youtube/helper/utils.py:57]: `Replies` needs to be localized

File: resources/lib/youtube_plugin/youtube/helper/utils.py (hunk 1 above)
Comment (doko-desuka) [reply resources/lib/youtube_plugin/youtube/helper/utils.py]: Of course, I completely forgot about localization.

**Comment by doko-desuka** [comment]
@anxdpanic I added the localizations as you requested, if you think of anything else please let me know.  
I also changed some visual things to make it easier to use:  

ytComments2

The comment texts are now also added to the item labels to make some of them quicker to read when they're only a single line. It also compacts the "likes" and "replies" values into those colored labels.

@mrmvergeer thanks for looking into it. I noticed that for some reason I was setting `process_next_page=False`, which was causing this, so now it's fixed.  
But it's not that great anyway because the YouTube API only serves comments in reverse order: newest first, oldest last (this is different than what the website does, it shows oldest replies first, newest last). I filed a bug report, I hope they change this in the future and add the same sorting as the website uses.

**Comment by doko-desuka** [comment]
Minor change, removed the quotes from the text (it was just adding noise).

ytComments3

**Comment by anxdpanic** [comment]
@doko-desuka Thanks again!, it all looks good to me now.

**Comment by anxdpanic** [comment]
added to changelog in #789

**Comment by doko-desuka** [comment]
@anxdpanic np, thanks for the quick review

**Comment by anxdpanic** [comment]
Included in beta1, https://github.com/jdf76/plugin.video.youtube/releases/tag/6.7.1-dev

**Pull request merged**